### ISSUE TITLE
Refactoring of Multi Node Batch Normalization in ChainerMN

### ISSUE THREAD

**Issue by shu65** [chainer/chainer#6533]
There are many parts where Multi Node Batch Normalization in ChainerMN overlaps with BatchNormalization in Chainer.
So, we should refactor multi node batch normalization.

ToDo:

- [ ] add args for comm and backend to BatchNormalization in functions of Chainer
- [ ] use BatchNormalization in functions of Chainer in Multi Node Batch Normalization.
